libs/pages/settings/src/lib/feature/page-organization-billing-summary-feature/show-usage-modal-feature/show-usage-modal-feature.tsx

```diff
@@ -0,0 +1,76 @@
+import { Organization, type OrganizationCurrentCost } from 'qovery-typescript-axios'
+import { useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useGenerateBillingUsageReport, useOrganization } from '@qovery/domains/organizations/feature'
+import { useModal } from '@qovery/shared/ui'
+import ShowUsageModal, {
+  getReportPeriods,
+} from '../../../ui/page-organization-billing-summary/show-usage-modal/show-usage-modal'
+import ShowUsageValueModal from '../../../ui/page-organization-billing-summary/show-usage-value-modal/show-usage-value-modal'
+
+export interface ShowUsageModalFeatureProps {
+  organizationId: string
+  currentCost: OrganizationCurrentCost
+}
+
+export function ShowUsageModalFeature({ organizationId, currentCost }: ShowUsageModalFeatureProps) {
+  const methods = useForm<{ expires: number; report_period: string }>({
+    defaultValues: {
+      expires: 24,
+      report_period: 'current_month',
+    },
+    mode: 'all',
+  })
+
+  const { data: organization } = useOrganization({ organizationId })
+  const { mutateAsync: usageBillingReport } = useGenerateBillingUsageReport()
+  const { openModal, closeModal } = useModal()
+  const [loading, setLoading] = useState(false)
```

Comment:
You can remove this `useState` to manage the loading and use directly `isLoading` param by the `useGenerateBillingUsageReport`